spinnaker/spinnaker_system/google_http_lb_upsert_test.py

```diff
@@ -0,0 +1,894 @@
+
+
+  def _make_contract_clauses(self, contract_builder, upsert):
+    '''Add the proper predicates to the contract builder for a given
+    upsert description.
+    '''
+    host_rules = upsert['hostRules'] # Host rules will be distinct.
+    backend_services = [upsert['defaultService']]
+    for host_rule in host_rules:
+      path_matcher = host_rule['pathMatcher']
+      backend_services.append(path_matcher['defaultService'])
+      for path_rule in path_matcher['pathRules']:
+        backend_services.append(path_rule['backendService'])
+    health_checks = [service['healthCheck'] for service in backend_services]
+
+    hc_clause_builder = contract_builder.new_clause_builder('Health Checks Created',
+                                                            retryable_for_secs=30)
+    hc_clause_builder = hc_clause_builder.list_resource('httpHealthChecks')
+    for hc in health_checks:
+      hc_clause_builder.contains_pred_list(
+        [
+          jp.PathEqPredicate('name', hc['name']),
+          jp.PathEqPredicate('requestPath', hc['requestPath']),
+          jp.PathEqPredicate('port', hc['port'])
+        ]
+      )
+
+    bs_clause_builder = contract_builder.new_clause_builder('Backend Services Created',
+                                                            retryable_for_secs=30)
+    bs_clause_builder = bs_clause_builder.list_resource('backendServices')
+    for bs in backend_services:
+      bs_clause_builder.contains_pred_list(
+        [
+         jp.PathEqPredicate('name', bs['name']),
+         jp.PathEqPredicate('portName', 'http'),
+         jp.PathContainsPredicate('healthChecks[0]', bs['healthCheck']['name'])
+        ]
+      )
+
+    url_map_clause_builder = contract_builder.new_clause_builder('Url Map Created',
+                                                                 retryable_for_secs=30)
+    url_map_clause_builder = url_map_clause_builder.list_resource('urlMaps')
+    for hr in host_rules:
+      pm = hr['pathMatcher']
+      pred_list = [
+         jp.PathEqPredicate('name', self.__lb_name),
+         jp.PathEqPredicate('defaultService',
+                                  self._make_bs_link(upsert['defaultService']['name'])),
+         jp.PathContainsPredicate('pathMatchers/defaultService',
+                                  self._make_bs_link(pm['defaultService']['name'])),
+      ]
+      for host in hr['hostPatterns']:
+        pred_list.append(
+         jp.PathContainsPredicate('hostRules/hosts', host),
+        )
+      for pr in pm['pathRules']:
+        pred_list.append(
+         jp.PathContainsPredicate('pathMatchers/pathRules/service',
+                                  self._make_bs_link(pr['backendService']['name'])),
+        )
+        for path in pr['paths']:
+          pred_list.append(
+            jp.PathContainsPredicate('pathMatchers/pathRules/paths', path),
+          )
+      url_map_clause_builder.contains_pred_list(
+        pred_list
```

Comment:
The wrap here isnt necessary, especially if you combine the clause_builder statements together into one statement.